test/unit/workspace/datastore_test.py

```diff
@@ -335,4 +338,99 @@ def test_get_resource_with_nonexistent_resource_fails(self):
         self.assertRaises(KeyError, self.fetcher.get_resource, res)
 
 
+def test_get_zipfile_resource_retry(mocker):
+    ds = datastore.Datastore()
+    not_a_zipfile = b"aaa"
+    ds.get_unique_resource = mocker.MagicMock(return_value=not_a_zipfile)
+    sleep_mock = mocker.MagicMock()
+    with mocker.patch("time.sleep", sleep_mock), pytest.raises(zipfile.BadZipFile):
+        ds.get_zipfile_resource("test_dataset")
+
+    sleep_mock.assert_has_calls([mocker.call(2**x) for x in range(5)])
```

Comment:
These assertions of "retried several times" seem like they're just re-testing the retry logic we've already tested - I'm happy to take them out.